WickedEngine/wiGraphicsDevice_Vulkan.h

```diff
@@ -595,346 +595,105 @@ namespace wi::graphics
 			// Deferred destroy of resources that the GPU is already finished with:
 			void Update(uint64_t FRAMECOUNT, uint32_t BUFFERCOUNT)
 			{
-				destroylocker.lock();
+				const uint64_t DEADLINE = FRAMECOUNT - BUFFERCOUNT;
```

Comment:
If FRAMECOUNT is less than BUFFERCOUNT this would cause negative uint64_t which will be a huge number. Then the `queue.front().second < DEADLINE` always passes and resource could be deleted while in use. I think this is a bug opportunity.

Reply:
Oops, you're absolutely right. I shouldn't do last minute "optimizations" late at night.